app/controllers/application_controller.rb

```diff
@@ -5,6 +5,133 @@ class ApplicationController < Sinatra::Base
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "secret"
+  end
+get '/tweets/new' do 
+    if Helpers.is_logged_in?(session)
+      erb :'/tweets/create_tweet'
+    else 
+      redirect '/login'
+    end
+  end
+
+  post '/tweets' do
+    if params[:content] == ""
+      redirect :'/tweets/new'
+    else
+      user = User.find_by_id(session[:user_id])
+      Tweet.create(content: params[:content], user_id: user.id)
+    end
+
+  end
+
+  get '/tweets' do 
+
+    if session[:user_id]
+      @tweets = Tweet.all
+      erb :"tweets/tweets"
+    else
+      redirect '/login'
+    end
   end
 
+  get '/users/:username' do
+    @user = User.find_by(:username => params[:username])
+    erb :'/users/show'
+  end
+
+  get '/tweets/:id' do
+    if Helpers.is_logged_in?(session)
+      @tweet = Tweet.find_by_id(params[:id])
+      erb :'tweets/show_tweet'
+    else
+      redirect to '/login'
+    end
+  end
+
+  get '/tweets/:id/edit' do 
+    if Helpers.is_logged_in?(session) 
+      @tweet = Tweet.find_by_id(params[:id])
+      if @tweet.user_id == session[:user_id]
+        erb :'/tweets/edit_tweet'
+      else
+        redirect to '/tweets'
+      end
+    else
+      redirect to '/login'
+    end
+  end
+
+  post '/tweets/:id' do
+    if params[:content] != ""
+      @tweet = Tweet.find_by_id(params[:id])
+      @tweet.content = params[:content]
+      @tweet.save
+    else
+      redirect to "/tweets/#{params[:id]}/edit"
+    end
+  end
+
+  post '/tweets/:id/delete' do 
+    if session[:user_id] 
+      @tweet = Tweet.find_by_id(params[:id])
+      if @tweet.user_id == session[:user_id]
```

Comment:
Prefer to find tweets that are already only associated to the `current_user` than to manually check the `user_id` foreign key of the tweet against the `current_user.id`

```ruby
tweet = current_user.tweets.find_by(:id => params[:id])
if tweet && tweet.destroy
  redirect "/tweets"
else
  redirect "/tweets/#{tweet.id}"
end
```

We're using the association collection method provided by [`has_many`](http://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html#method-i-has_many) to query the user's tweets. 

Note the usage of [`find_by`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find-by) vs [`find`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find). Had we used `find` here, ActiveRecord would raise an error in the event that we could not find a tweet associated with this user. `find_by` will allow this action to fail gracefully, simply returning nil if the user is trying to delete a tweet that they do not have access to.